nixos/modules/services/web-servers/nginx/location-options.nix

```diff
@@ -9,6 +9,31 @@ with lib;
 
 {
   options = {
+    basicAuth = mkOption {
+      type = types.attrsOf types.str;
+      default = {};
+      example = literalExample ''
+        {
+          user = "password";
+        };
+      '';
+      description = ''
+        Basic Auth protection for a vhost.
+
+        WARNING: This is implemented to store the password in plain text in the
+        nix store.
+      '';
+    };
+
+    basicAuthFile = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = ''
+        Basic Auth password file for a vhost.
+        Can be created via: <command>htpasswd -c &lt;filename&gt; &lt;username&gt;</command>
```

Comment:
Maybe it should be mentioned that the hashes htpasswd produces by default and all the hashes nginx actually understands aren't really considered good for storing passwords anymore.

Reply:
Possibly so, what would you like the wording to be? Note: these options and their text mirror the vhost-options.nix content.

Reply:
Right. I mean, in an ideal world I would like this to say "use `htpasswd -B` to create bcrypt hashed passwords", but nginx doesn't support those.

So maybe just something along the lines of: "Warning: The generate file contains the users' passwords in a non-cryptographically-securely hashed way"